src/main/java/marquez/service/DatasourceService.java

```diff
@@ -0,0 +1,43 @@
+package marquez.service;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import marquez.common.models.ConnectionUrl;
+import marquez.common.models.DatasourceName;
+import marquez.db.DatasourceDao;
+import marquez.db.models.DatasourceRow;
+import marquez.service.exceptions.UnexpectedException;
+import marquez.service.mappers.DatasourceMapper;
+import marquez.service.models.Datasource;
+
+@Slf4j
+public class DatasourceService {
+  private final DatasourceDao dataSourceDao;
+
+  public DatasourceService(@NonNull final DatasourceDao dataSourceDao) {
```

Comment:
what about `get()` for looking up a single `Datasource`?

Reply:
Please see PRs #394, #387, and #384, where this is fixed.